docs/content/sips/006-spin-plugins.md

```diff
@@ -0,0 +1,233 @@
+
+A [`spin-plugins` repository](#centralized-plugin-manifest-repository) will act as an inventory of available plugins, made by both Spin maintainers and the community. In the repository, a plugin will be defined by a [JSON Spin plugin manifest](#spin-plugin-manifest). Spin will pull down this manifest during installation, which will instruct it on where to find the plugin binary, version, platform compatibility, and more.
+
+### Usage
+
+The `spin plugin` command will have three sub-commands.
+
+```bash
+Commands for working with Spin plugins
+
+USAGE:
+    spin plugin <SUBCOMMAND>
+
+SUBCOMMANDS:
+    install      Install plugin as described by a remote or local plugin manifest
+    uninstall    Uninstall a plugin
+	update		 Update one or all plugins to the latest or specified version 
+```
+
+**`spin plugin install`**
+
+The `spin plugin install` subcommand installs a plugin named `$name`. By default, it will look for a plugin manifest named `$name.json` in the `spin-plugin` repository; however, it can be directed to use a local manifest or one at a different remote location using the `--file` or `--url` flag, respectively.
+
+```bash
+Install a Spin plugin using a plugin manifest file. 
+By default, looks for the plugin manifest named <name>.json
+in the Spin plugins repository https://github.com/fermyon/spin-plugins
+
+USAGE:
+    spin plugin install <name>
+
+OPTIONS:
+    -f, --file                       Path to local plugin manifest
+	-u, --url                        Address of remote plugin manifest
+    -v, --version                    Desired version to update plugin to. Defaults to latest.
+    -s, --skip-version-compat-check  Instructs Spin to install the plugin even 
+                                     if its manifest specifies that it is 
+                                     incompatible with the current version of Spin
+```
+
+If the manifest is found, Spin will check that the plugin is compatible with the current OS, platform, and version of Spin. If so, before installing the plugin, Spin will prompt the user as to whether to trust the source. For example, the following prompt would be displayed for a plugin named `deploy` with an Apache 2 license and hosted at [`https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz`](https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz):
+
+```bash
+Installing plugin deploy with license Apache 2.0 from https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz
+For more information, reference the plugin metadata at `https://github.com/fermyon/spin-plugins/plugin-manifests/deploy.json`.
+Are you sure you want to proceed? ('yes'/'no') (default: no)
+```
+
+The plugin will only be installed if a user enters `yes`. Otherwise, the command exits. 
```

Comment:
Very tactical detail, but there should be an option to bypass the prompt for CI scenarios.

Reply:
Good point. Maybe a `-y` flag as in `apt`?

Reply:
added in d75cfed287ce